### ISSUE TITLE
[Feature Request] Support .boxer bundles.

### ISSUE THREAD

**Issue by eduo** [litchie/dospad#153]
What the subject says.

Boxer has supported a .boxer format similar to the .idos format for some time. It's not exactly the same (files are dumped straight into the folder rather than in a subfolder).

While the file structure is not that different, what may be different is the defintion of bundles with extension .floppy, .cdrom and .harddisk which are mounted by DOSBox as part of opening the Package. If there are ISOs they're also mounted (I think iDOS currently doesn't support opening ISOs or BIN/CUE even though DOSBox does and can be done from the command line). 

Here's the author explaining it a bit: https://www.vogons.org/viewtopic.php?p=148547#p148547

(Boxer has many other enhancements to make autodetection and autoconfig of specific packages better for new users, but that's a separate thing you may already be aware of. [Boxer's most current fork is MaddTheSane's](https://github.com/MaddTheSane/Boxer) ([beta binaries](https://boxer.thec0de.com/)), which I'm sure you're aware of already)

**Comment by litchie** [comment]
Interesting idea to auto mount floppy and cdrom folders. ISO mount is supported, but not BIN/CUE.

**Comment by litchie** [comment]
Previously:  #135
If you can share a few boxer packages with me, it would be great: idos@litchie.com

**Comment by eduo** [comment]
> Interesting idea to auto mount floppy and cdrom folders. ISO mount is supported, but not BIN/CUE.

Maybe I haven't found how to mount ISO from the GUI. I usually do it in the command line but no option I select allows me to choose an ISO (I can choose folders to mount as drives, though, which is similar to the .harddisk and .cdrom bundles from boxer).

BIN/CUE would be nice as well as ISO. ZIP is common as well, I guess.

Something missing is documentation of what format is a "disk image file" (as labeled in the GUI, since ISO doesn't seem to work).

**Comment by eduo** [comment]
> Previously: #135 If you can share a few boxer packages with me, it would be great: [idos@litchie.com](mailto:idos@litchie.com)

Sure thing. I will send you a couple of icloud share links via DM in twitter. I will send one with .harddisk and one with .cdrom. I don't have one with .floppy but I could build it if you wish. 

I'll also share a third one that shows an icon, but I understand it uses the MacOS functionality for custom icons and as such it's harder to translate to iOS what has no such concept (although it could convert the resource fork on the fly when first opened). Boxer allows dropping artwork while building the game, but this then uses system configuration do add the invisible Icon resource (I haven't checked how iDOS does it, I assume it uses iOS file system calls to define an icon with an image as preview).

These affordances were added in version 0.65 of Boxer: https://www.vogons.org/viewtopic.php?t=17834

In the demo boxer volumes there are examples of .harddisk in Ultima Underworld Demo and X-COM Demo:

https://github.com/search?q=repo%3AMaddTheSane%2FBoxer+.harddisk&type=code
(Also available in the website: https://boxer.thec0de.com/games.html )

According to the source .cdmedia is also supported, which I didn't know. It seems this is a bundle of BIN/CUE in a single folder for multi-CD.

Lastly: Boxer allows for a "Documentation" folder containing manuals or cheat sheets or whatnot, accessible as a floating window during emulation. It is accessibly within the application itself but I have also provided it as a ZIP.

Boxer was a very mature, very friendly dosbox front-end that sadly got forgotten for a while, but it has many great ideas that have popped up afterwards in other places. If I remember correctly Boxer became the basis for how GOG distributes legacy DOS games but I may be confusing it with something else.
